### PR TITLE
Edit delete command

### LINKED ISSUE #54: Delete multiple contacts by parameters

### PR DESCRIPTION
**Pull request by leroytan**
Close #54

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java [hunk 1]
```diff
@@ -24,25 +27,49 @@ public class DeleteCommand extends Command {
...
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
+        Person personToDelete = null;
         List<Person> lastShownList = model.getFilteredPersonList();
-
-        if (targetIndex.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        //reverse order to prevent index out of bounds
+        targetIndexes.sort(Comparator.comparingInt(Index::getZeroBased).reversed());
```
Comment (anselmlong) [inline src/main/java/seedu/address/logic/commands/DeleteCommand.java:62]: Why must we do this? Can you explain a bit more in the comments?

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java [hunk 2]
```diff
@@ -24,25 +27,49 @@ public class DeleteCommand extends Command {
...
+    /**
+     * Creates a DeleteCommand to delete the person at the specified {@code Index}.
+     */
+    public DeleteCommand(Index targetindex) {
+        this.targetIndexes = new ArrayList<>();
+        targetIndexes.add(targetindex);
     }
 
+    /**
+     * Creates a DeleteCommand to delete the person at the specified {@code Indexes}.
```
Comment (anselmlong) [inline src/main/java/seedu/address/logic/commands/DeleteCommand.java:43]: Javadocs here could be clearer that you are deleting for multiple people!

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 3]
```diff
@@ -37,6 +41,28 @@ public static Index parseIndex(String oneBasedIndex) throws ParseException {
         return Index.fromOneBased(Integer.parseInt(trimmedIndex));
     }
 
+    /**
+     * Parses {@code oneBasedIndexes} into a list of {@code Index} and returns it. Leading and trailing whitespaces will
+     * be trimmed.
+     *
+     * @throws ParseException if the specified index is invalid (not non-zero unsigned integer).
+     */
+    public static ArrayList<Index> parseMultipleIndexSeperatedByComma(String oneBasedIndexes) throws ParseException {
```
Comment (anselmlong) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:50]: Small nit but you could name this `parseCommaSeperatedIndexes` for clarity?

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 4]
```diff
@@ -37,6 +41,28 @@ public static Index parseIndex(String oneBasedIndex) throws ParseException {
...
+        String trimmedIndexes = oneBasedIndexes.trim();
+        if (trimmedIndexes.endsWith(",")) {
+            throw new ParseException(MESSAGE_INCOMPLETE_INDEX);
```
Comment (anselmlong) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:53]: Good error checking

File: src/test/java/seedu/address/logic/parser/DeleteCommandParserTest.java [hunk 5]
```diff
@@ -22,13 +27,25 @@ public class DeleteCommandParserTest {
...
+        ArrayList<Index> targetIndexes = new ArrayList<>(Arrays.asList(INDEX_FIRST_PERSON, INDEX_SECOND_PERSON));
+        assertParseSuccess(parser, " " + PREFIX_ID + "1, 2", new DeleteCommand(targetIndexes));
+    }
+
+    @Test
+    public void parse_invalidMultipleArgs_throwsParseException() {
+        assertParseFailure(parser, " " + PREFIX_ID + "1, a",
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteCommand.MESSAGE_USAGE));
+    }
 }
```
Comment (anselmlong) [inline src/test/java/seedu/address/logic/parser/DeleteCommandParserTest.java:51]: Good test cases.

**Review by anselmlong — COMMENTED** [review]
LGTM but some minor documentation or code clarity issues.

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java (hunk 1 above)
Comment (leroytan) [reply src/main/java/seedu/address/logic/commands/DeleteCommand.java:62]: Updated to explain more. Thanks!

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java (hunk 2 above)
Comment (leroytan) [reply src/main/java/seedu/address/logic/commands/DeleteCommand.java:43]: Updated to improve clarity!

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 3 above)
Comment (leroytan) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java:50]: Change method name to `parseCommaSeparatedIndexes` for clarity!

**Pull request merged**